File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        int operations = 0;
        map<int, int> hash;
        for (int& n : nums) {
            hash[n]++;
        }
        for (auto it : hash) {
            int curr = it.first;
            int comp = k - curr;
            if (curr == comp) {
                if (hash[curr] > 1) {
                    operations += hash[curr] / 2;
                    hash[curr] = 0;
                }
            } else {
                if (hash[curr] > 0 && hash[comp] > 0) {
                    operations += min(hash[curr], hash[comp]);
                    hash[curr] = hash[comp] = 0;
                }
            }
        }
        return operations;
    }
};

File: 1679-max-number-of-k-sum-pairs/1679-max-number-of-k-sum-pairs.java
class Solution {
    public int maxOperations(int[] nums, int k) {
        Arrays.sort(nums);
        int low = 0, high = nums.length - 1, result = 0;
        while (low < high) {
            if (nums[low] + nums[high] == k) {
                low++;
                high--;
                result++;
            } else if (nums[low] + nums[high] < k) {
                low++;
            } else {
                high--;
            }
        }
        return result;
    }
}